utility/src/test/java/org/oppia/android/util/math/FloatExtensionsTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.oppia.android.util.math
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.annotation.LooperMode
+
+/** Tests for [Float] and [Double] extensions. */
+// FunctionName: test names are conventionally named with underscores.
+@Suppress("FunctionName")
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+class FloatExtensionsTest {
+
+  @Test
+  fun testFloat_approximatelyEquals_bothZero_returnsTrue() {
+    val leftFloat = 0f
+    val rightFloat = 0f
+
+    val result = leftFloat.approximatelyEquals(rightFloat)
+
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun testFloat_approximatelyEquals_sameNonZeroValue_returnsTrue() {
+    val leftFloat = 1.2f
+    val rightFloat = 1.2f
+
+    val result = leftFloat.approximatelyEquals(rightFloat)
+
+    assertThat(result).isTrue()
+  }
+
+  @Test
+  fun testFloat_approximatelyEquals_nonZeroValues_withinInterval_returnsTrue() {
+    val leftFloat = 1.2f
+    val rightFloat = leftFloat + FLOAT_EQUALITY_INTERVAL.toFloat() / 10f
+
+    val result = leftFloat.approximatelyEquals(rightFloat)
+
+    // Verify that they are approximately equal, but not actually the same float.
+    assertThat(result).isTrue()
+    assertThat(leftFloat).isNotEqualTo(rightFloat)
+  }
+
+  @Test
+  fun testFloat_approximatelyEquals_nonZeroValues_outsideInterval_returnsFalse() {
+    val leftFloat = 1.2f
+    val rightFloat = leftFloat + FLOAT_EQUALITY_INTERVAL.toFloat() * 2f
+
+    val result = leftFloat.approximatelyEquals(rightFloat)
+
+    assertThat(result).isFalse()
+  }
+
+  @Test
+  fun testFloat_approximatelyEquals_nonZeroValues_veryDifferent_returnsFalse() {
+    val leftFloat = 1.2f
+    val rightFloat = 7.3f
```

Comment:
Do we also need a test case where:
```kt
val leftFloat = 0f
val rightFloat = 7.3f
```
returns false?

Reply:
Sure that seems sensible--done.